source/common/secret/secret_manager_impl.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <unordered_map>
+
+#include "envoy/secret/secret.h"
+#include "envoy/secret/secret_manager.h"
+
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Secret {
+
+class SecretManagerImpl : public SecretManager, Logger::Loggable<Logger::Id::upstream> {
+public:
+  SecretManagerImpl(){};
+
+  void addOrUpdateSecret(SecretSharedPtr secret) override;
+  const SecretSharedPtr findSecret(const std::string& name) const override;
+
+private:
+  typedef std::unordered_map<std::string, SecretSharedPtr> SecretSharedPtrMap;
```

Comment:
can Secret be stored as unique_ptr?

Reply:
Secret needs to be shared pointer to typecast with std::dynamic_pointer_cast.